### PR TITLE
Booze cures anxiety (and other ethanol diseases)

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
Viruses cured by ethanol now accept any form of booze.

## Why It's Good For The Game
It's much more entertaining to tell the crew "Go get drunk at the bar, doctor's orders!" than "There are ethanol pills in the fridge."

## Testing
Gave myself severe anxiety.  Drank coffee.  It didn't help.  Drank wine.  Cured.
Gave myself the flu.  Injected spaceacillin.  Cured.
Gave myself GBS.  Drank sulfur alone.  Didn't help.  Drank diphenhydramine + sulfur.  Cured
New character.  Gave myself severe anxiety.  Drank pure ethanol.  Cured.

## Changelog
:cl:
tweak: Diseases cured by ethanol now accept any form of booze, not just pure ethanol.
/:cl:

### CODE REVIEW

**Comment by CinnamonSnowball** [comment]
This might be better fixed by changing how Severe Anxiety checks for cures, rather than altering the cure checking code for *all* viruses

**Comment by FunnyMan3595** [comment]
Severe anxiety isn't the only disease that's cured by ethanol, it's also part of the cures list for all advanced diseases:
https://github.com/ParadiseSS13/Paradise/blob/14baa57030ec829665b9164c7df8848e000acd8e/code/datums/diseases/advance/advance.dm#L14

File: code/datums/diseases/_disease.dm [hunk 1]
```diff
@@ -108,12 +108,22 @@ GLOBAL_LIST_INIT(diseases, subtypesof(/datum/disease))
...
+			var/found_booze = FALSE
+			for(var/datum/reagent/consumable/ethanol/booze in affected_mob.reagents.reagent_list)
+				found_booze = TRUE
+				break
+			if(found_booze)
+				cures_found++
+		else if(affected_mob.reagents.has_reagent(C_id))
+			cures_found++
+
+	if(needs_all_cures && cures_found < cures.len)
```
Comment (lewcc) [inline code/datums/diseases/_disease.dm:123]:
```suggestion
	if(needs_all_cures && cures_found < length(cures))
```

File: code/datums/diseases/_disease.dm [hunk 2]
```diff
@@ -108,12 +108,22 @@ GLOBAL_LIST_INIT(diseases, subtypesof(/datum/disease))
...
-			.--
-	if(!. || (needs_all_cures && . < cures.len))
-		return 0
+		if(C_id == "ethanol")
+			var/found_booze = FALSE
+			for(var/datum/reagent/consumable/ethanol/booze in affected_mob.reagents.reagent_list)
+				found_booze = TRUE
+				break
+			if(found_booze)
+				cures_found++
```
Comment (lewcc) [inline code/datums/diseases/_disease.dm:119]:
```suggestion
			for(var/datum/reagent/consumable/ethanol/booze in affected_mob.reagents.reagent_list)
				cures_found++
				break
```

**Review by lewcc — APPROVED** [review]
LGTM

**Review by Contrabang — APPROVED** [review]

**Pull request merged**